### PR TITLE
Edit feature

### PR DESCRIPTION
**Pull request by vineshgobin**
Using PUT http method to modify data then updating the database.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,41 +1,64 @@
...
+  getEmployees,
+  createEmployee,
+  deleteEmployee,
+  updateEmployeeConst,
+} from "./api/EmployeeService";
 
 const App = () => {
-  const initialFormState = { id: null, name: "", username: "" }
-  const [currentUser, setCurrentUser] = useState(initialFormState)
+  const initialFormState = {
```
Comment (gilsantosjulian) [inline src/App.js]: This `initialFormState ` object is this needed?

File: src/App.js [hunk 2]
```diff
@@ -1,41 +1,64 @@
...
+    id: null,
+    firstName: "",
+    middle: "",
+    lastName: "",
+  };
+  //const [currentUser, setCurrentUser] = useState(initialFormState);
```
Comment (gilsantosjulian) [inline src/App.js]: Remove this and the rest comments that are not needed

File: src/App.js [hunk 3]
```diff
@@ -1,41 +1,64 @@
...
+    const employees = await getEmployees();
+    setEmployees(employees);
+  }, []);
 
-  const editRow = (user) => {
-    setEditing(true)
-    setCurrentUser(user)
-  }
+  const editRow = (employee) => {
+    console.log(employee);
```
Comment (gilsantosjulian) [inline src/App.js]: Remove it

File: src/App.js [hunk 4]
```diff
@@ -1,41 +1,64 @@
...
+    setEmployee(employee);
+  };
 
   const addEmployee = (employee) => {
-    console.log({ employee })
-    employee.id = employees.length + 1
-    setEmployees([...employees, employee])
-    createEmployee({ body: employee })
-  }
+    console.log({ employee });
```
Comment (gilsantosjulian) [inline src/App.js:33]: Remove it

File: src/App.js [hunk 5]
```diff
@@ -1,41 +1,64 @@
...
+
+  const updateEmployee = (id, updatedEmployee) => {
+    setEditing(false);
+    setEmployee(
+      employees.map((employee) =>
+        employee.id === id ? updatedEmployee : employee
+      )
+    );
+    updateEmployeeConst({ id: id, body: updatedEmployee });
+    console.log({ updatedEmployee });
```
Comment (gilsantosjulian) [inline src/App.js]: Remove it

File: src/api/EmployeeService.js [hunk 6]
```diff
@@ -48,6 +48,22 @@ export const deleteEmployee = async (id) => {
   }
 };
 
+export const updateEmployeeConst = async ({ id, body }) => {
+  try {
+    const response = await fetch(
+      `https://apex.oracle.com/pls/apex/hoangtuanmai/api/employee/${id}`,
+      {
+        // const response = await fetch(`${API_URL}/employee`, {
```
Comment (gilsantosjulian) [inline src/api/EmployeeService.js]: Remove it

File: src/api/EmployeeService.js [hunk 7]
```diff
@@ -48,6 +48,22 @@ export const deleteEmployee = async (id) => {
...
+        body: JSON.stringify(body),
+      }
+    );
+    return response;
+  } catch (error) {
+    console.log({ error });
+  }
+};
+
 // export const updateEmployee = (id, firstname, middlename, lastname, soc, hiredate, salary, commissionpct, departmentcode, jodcode, managerid) => {
```
Comment (gilsantosjulian) [inline src/api/EmployeeService.js]: Delete this

File: src/forms/EditUser.js [hunk 8]
```diff
@@ -1,36 +1,49 @@
-import React, { useState, useEffect } from 'react'
```
Comment (gilsantosjulian) [inline src/forms/EditUser.js:1]: This file seems that is not necessary, can you check, and if not, you can remove it and all its imports

**Review by gilsantosjulian — APPROVED** [review]

File: src/App.js [hunk 9]
```diff
@@ -1,41 +1,64 @@
...
+  };
```
Comment (gilsantosjulian) [inline src/App.js:55]: Remove `console.log({ editing });`

Also can you change `CRUD App with Hooks` to `CRUD App - Oracle Database`

**Review by maituan94 — APPROVED** [review]

**Pull request merged**